### PR TITLE
Victory overhaul

### PR DESCRIPTION
**Pull request by silasary**
Just a small change to the way Victory locations are handled.

We no longer need to replace the original location with `__Manual Game Complete__` (Though it remains if the creator didn't make a victory location)

Instead, we simply place `__Victory__` on the json defined location directly.  

Also, I added the ability to define multiple victory locations, and an auto-yaml option to set your goal if you do so.

### CODE REVIEW

**Comment by FuzzyGamesOn** [comment]
This is awesome! I do want to test it out some to see if we can catch any issues before rolling it into an unstable. Changing the victory stuff makes me a bit nervous 😃

**Review by nicopop — CHANGES_REQUESTED** [review]
so far it work when I reAdded custom_victory_location
but I think victory locations should be removed/disabled/excluded if they are not the goal
it probably should be at least an option
judging by my spoiler logs non-filler items can be put in there and if your game is a one way to the victory location then you might be stuck or something idk,

also the selected goal's location still get renamed to "Victory! (seed finished)" in the client and UT stop working but just for that location
the other goal locations (unselected) get to keep there name and UT support
its seems to be a client problem not a you problem since in the spoiler log it uses the correct location name

File: src/Locations.py
Comment (silasary) [reply src/Locations.py:32]: Good catch

**Comment by silasary** [comment]
> also the selected goal's location still get renamed to "Victory! (seed finished)" in the client and UT stop working but just for that location

Yeah, the client stuff is a whole other issue, and not one I'm addressing with this PR.  Fixing the client side UI is an entirly separate mess, and I wanted to do it as two PRs

**Review by nicopop — APPROVED** [review]
the changes looks good!

**Review by FuzzyGamesOn — APPROVED** [review]
This all looks good. We can't merge this until the client is accounted for, since this PR would overwrite the `__Manual Game Complete__` location with the user-defined one. But once the client PR is made and approved, we can merge both. 😄

**Comment by silasary** [comment]
> We can't merge this until the client is accounted for, since this PR would overwrite the `__Manual Game Complete__` location with the user-defined one. But once the client PR is made and approved, we can merge both. 😄

Not true.  The client doesn't actually use the location at all.  I've completed games with this code using the latest stable client, and it all works exactly the same as it always has.

**Pull request merged**
